src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1694,6 +1696,9 @@ SingleShardSelectTaskList(Query *query, uint64 jobId, List *relationShardList,
 	StringInfo queryString = makeStringInfo();
 	List *relationRowLockList = NIL;
 
+	/* try to avoid doing many allocations during deparsing */
+	enlargeStringInfo(queryString, 500);
```

Comment:
`makeStringInfo()` initially allocates space for 1024 characters, so I think this should have no effect.

Reply:
That's a good point, thanks @pykello.

@marcocitus given that basic fast-path router queries generally less than < 1024 chars length, I'm planning to remove this commit. Let me know if you think otherwise.